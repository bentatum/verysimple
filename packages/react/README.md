# @verysimple/react

This is the `@verysimple/react` package