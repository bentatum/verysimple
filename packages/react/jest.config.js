module.exports = {
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect"],
  collectCoverage: true,
  coverageThreshold: {
    global: {
      statements: 95,
      branches: 95,
      functions: 95,
      lines: 95,
    },
  },
  coverageReporters: ["text-summary", "json-summary"],
  collectCoverageFrom: ["src/**/*.{ts,tsx}", "!**/*.d.ts"],
  coverageDirectory: "coverage",
  moduleNameMapper: {
    "@/test/(.*)": "<rootDir>/test/$1",
    "@/(.*)$": "<rootDir>/src/$1",
  },
  modulePathIgnorePatterns: ["<rootDir>/dist/"],
};
