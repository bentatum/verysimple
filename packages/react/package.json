{
  "name": "@verysimple/react",
  "version": "8.1.1",
  "description": "A very simple component library",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "author": "Ben Tatum <mail@benjtatum.com>",
  "license": "MIT",
  "repository": "https://github.com/bentatum/verysimple",
  "scripts": {
    "test": "jest",
    "prebuild": "rm -rf dist",
    "build": "tsc --project tsconfig.build.json && tsc-alias"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.20.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/inquirer": "latest",
    "@types/jest": "^29.2.3",
    "@types/react": "^18.0.21",
    "@types/testing-library__jest-dom": "^5.14.5",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "classnames": "^2.3.2",
    "framer-motion": "^10.15.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.1.8",
    "tsc-alias": "^1.8.2",
    "typescript": "^4.9.3"
  },
  "volta": {
    "node": "20.10.0",
    "yarn": "1.22.21"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.9",
    "@heroicons/react": "^2.0.13",
    "clsx": "^2.1.0",
    "framer-motion": "^10.15.0"
  }
}
