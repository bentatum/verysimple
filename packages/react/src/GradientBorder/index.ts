export { default } from './GradientBorder'
export * from './GradientBorder'