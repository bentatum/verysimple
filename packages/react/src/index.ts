export * from "./ButtonBase";
export * from "./Card";
export * from "./Dialog";
export * from "./FilledButtonBase";
export * from "./FilledIconButton";
export * from "./GradientBorder";
export * from "./helpers";
export * from "./IconButton";
export * from "./NormalIconButton";
export * from "./OutlinedButtonBase";
export * from "./OutlinedIconButton";
export * from "./Spinner";
export * from "./types";
