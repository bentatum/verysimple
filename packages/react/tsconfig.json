{
  "extends": "@verysimple/tsconfig/react-library.json",
  "compilerOptions": {
    "outDir": "./dist",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules"]
}
