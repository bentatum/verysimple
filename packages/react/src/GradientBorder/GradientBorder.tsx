import {
  gradientBorderStyles,
  GradientBorderStylesOptions,
} from "@/helpers/gradientBorderStyles";
import { ComponentPropsWithRef, ElementType, forwardRef } from "react";

export type GradientBorderSize = 1 | 2 | 4 | 8;

export interface GradientBorderProps
  extends ComponentPropsWithRef<any>,
    GradientBorderStylesOptions {
  as?: ElementType;
  borderColor?: string;
}

export const GradientBorder = forwardRef<HTMLDivElement, GradientBorderProps>(
  (
    {
      as,
      style,
      borderSize,
      borderColor,
      backgroundColor,
      startColor,
      stopColor,
      direction,
      ...props
    },
    ref
  ) => {
    const Component = as || "div";
    return (
      <Component
        {...props}
        ref={ref}
        style={{
          ...gradientBorderStyles({
            backgroundColor,
            startColor: borderColor ?? startColor,
            stopColor: borderColor ?? stopColor,
            borderSize,
            direction,
          }),
          ...style,
        }}
      />
    );
  }
);

GradientBorder.displayName = "GradientBorder";

export default GradientBorder;
