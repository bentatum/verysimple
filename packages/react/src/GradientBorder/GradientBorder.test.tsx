import { render } from "@testing-library/react";
import { createRef } from "react";
import GradientBorder from "./GradientBorder";

describe("GradientBorder", () => {
  it("should have a gradient border", () => {
    const { container } = render(<GradientBorder>GM</GradientBorder>);
    expect(container.firstChild).toHaveStyle({
      background:
        "linear-gradient(white, white) 0 0 / 100% 100% no-repeat, linear-gradient(to right, #ffd200, #fb923c, #ff00e5)",
      backgroundClip: "content-box, border-box",
    });
    expect(container).toMatchSnapshot();
  });
  describe("borderSize", () => {
    test("1", () => {
      const { container } = render(
        <GradientBorder borderSize={1}>GM</GradientBorder>
      );
      expect(container.firstChild).toHaveStyle("padding: 1px");
      expect(container).toMatchSnapshot();
    });
    test("2", () => {
      const { container } = render(
        <GradientBorder borderSize={2}>GM</GradientBorder>
      );
      expect(container.firstChild).toHaveStyle("padding: 2px");
      expect(container).toMatchSnapshot();
    });
    test("4", () => {
      const { container } = render(
        <GradientBorder borderSize={4}>GM</GradientBorder>
      );
      expect(container.firstChild).toHaveStyle("padding: 4px");
      expect(container).toMatchSnapshot();
    });
    test("8", () => {
      const { container } = render(
        <GradientBorder borderSize={8}>GM</GradientBorder>
      );
      expect(container.firstChild).toHaveStyle("padding: 8px");
      expect(container).toMatchSnapshot();
    });
  });
  it("should have a direction prop", () => {
    const { container } = render(
      <GradientBorder direction="bottom">GM</GradientBorder>
    );
    expect(container.firstChild).toHaveStyle({
      background:
        "linear-gradient(white, white) 0 0 / 100% 100% no-repeat, linear-gradient(to bottom, #ffd200, #fb923c, #ff00e5)",
      backgroundClip: "content-box, border-box",
    });
    expect(container).toMatchSnapshot();
  });

  it("forwards the ref", () => {
    const ref = createRef<HTMLDivElement>();
    render(<GradientBorder ref={ref}>test</GradientBorder>);
    expect(ref.current).toBeInTheDocument();
  });

  it("has a display name", () => {
    expect(GradientBorder.displayName).toBe("GradientBorder");
  });
});
