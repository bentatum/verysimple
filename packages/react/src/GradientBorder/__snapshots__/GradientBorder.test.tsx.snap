// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientBorder borderSize 1 1`] = `
<div>
  <div
    style="background-clip: content-box, border-box; padding: 1px;"
  >
    GM
  </div>
</div>
`;

exports[`GradientBorder borderSize 2 1`] = `
<div>
  <div
    style="background-clip: content-box, border-box; padding: 2px;"
  >
    GM
  </div>
</div>
`;

exports[`GradientBorder borderSize 4 1`] = `
<div>
  <div
    style="background-clip: content-box, border-box; padding: 4px;"
  >
    GM
  </div>
</div>
`;

exports[`GradientBorder borderSize 8 1`] = `
<div>
  <div
    style="background-clip: content-box, border-box; padding: 8px;"
  >
    GM
  </div>
</div>
`;

exports[`GradientBorder should have a direction prop 1`] = `
<div>
  <div
    style="background-clip: content-box, border-box; padding: 1px;"
  >
    GM
  </div>
</div>
`;

exports[`GradientBorder should have a gradient border 1`] = `
<div>
  <div
    style="background-clip: content-box, border-box; padding: 1px;"
  >
    GM
  </div>
</div>
`;
